### DISCUSSION TITLE
security worries

### DISCUSSION THREAD

**Discussion by ARCIST-AI** [EtiennePerot/open-webui-code-execution#6] [Q&A]
Hi Étienne,
god i love that name. 
Anyway, i want  to use the code function but i think i need to do it the hard way because Firefox keeps tellin me my  knickers are down because all my local addresses are wide open to the world. So unless i can find a fix for that short of having my own intranet, i need to  do things the hard way.
The second problem is that my Dockerised Openwebui is inside a new consumer unit called Zima Cube,  which just shows me a settings pane. Could you suggest how i could set it up from there?
<img width="1287" alt="Screenshot 2024-09-08 at 22 02 22" src="https://github.com/user-attachments/assets/b69e7d44-69fe-4664-81a6-5ec0e8938069">
<img width="1287" alt="Screenshot 2024-09-08 at 22 03 41" src="https://github.com/user-attachments/assets/84f69347-71f9-4e29-b84f-45caf753db54">
<img width="1287" alt="Screenshot 2024-09-08 at 22 04 25" src="https://github.com/user-attachments/assets/2593a3e1-8182-492b-a69b-409b8534ecf7">
@EtiennePerot

**Comment by EtiennePerot** [comment]
The Zima Cube looks pretty slick NGL.

But yeah, looks like its web UI doesn't have a "privileged" box. Or perhaps clicking this label will do something?

<img width="1287" alt="365467287-84f69347-71f9-4e29-b84f-45caf753db54" src="https://github.com/user-attachments/assets/4a15218d-2f8b-4d3c-b8ba-a4149a0af624">

Others who have tried to set things up "the hard way" ran into various permission issues that `--privileged=true` seems to consistently fix; see #2 and #3. In any case, I am in the middle of testing a version that should require fewer permissions, less mucking around in the settings, and provide better debug data when something is off about the setup it's running in. Please hold for now.

**Reply by EtiennePerot** [reply]
I uploaded a new version of the code runner tool and function that includes a lot more debug logging. It also has more a elaborate self-test mode.

The **tool** is available [here](https://github.com/EtiennePerot/open-webui-code-execution/blob/dev/open-webui/tools/run_code.py) and the **function** is available [here](https://github.com/EtiennePerot/open-webui-code-execution/blob/dev/open-webui/functions/run_code.py).

If you can, please open a shell in the container, and run the tool's self-test mode:

```
python3 path/to/tools/run_code.py --self_test
```

If that fails, also add `--debug` to the above command-line, and it should generate more debug info.

If it works, then it should also work in Open WebUI!
